mmengine/infer/infer.py

```diff
@@ -156,22 +159,23 @@ def __init__(self,
             cfg = copy.deepcopy(model)
         elif isinstance(model, dict):
             cfg = copy.deepcopy(ConfigDict(model))
+        elif model is None:
+            assert weights is not None, (
+                'If model is None, the configuration of model must be read '
+                'from weights')
```

Comment:
```suggestion
                'If model is None, the weights must be specified and contain a config string because the configuration of model must be read '
                'from it')
```